component/src/main/java/org/wso2/carbon/identity/authenticator/nuxeo/NuxeoAuthenticator.java

```diff
@@ -0,0 +1,404 @@
+        } catch (OAuthProblemException e) {
+            throw new AuthenticationFailedException("Exception while getting the oAuth code from HttpServletRequest "
+                    + "for the client: " + clientId + ". Check whether you used the client_Secret value that " +
+                    "associate with clien_ID: " + clientId, e);
+        }
+    }
+
+    /**
+     * Get the OAuth response for access token.
+     *
+     * @param oAuthClient   the OAuthClient.
+     * @param accessRequest the AccessRequest.
+     * @return Response for access token from service provider.
+     */
+    private OAuthClientResponse getResponseOauth(OAuthClient oAuthClient, OAuthClientRequest accessRequest)
+            throws AuthenticationFailedException {
+
+        OAuthClientResponse oAuthResponse;
+        try {
+            oAuthResponse = oAuthClient.accessToken(accessRequest);
+        } catch (OAuthProblemException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Exception occurred from the invalid client_ID or Client_Secret: "
+                        + accessRequest.getBody() + " while getting the OAuth response for access token.");
+            }
+            throw new AuthenticationFailedException("Exception occurred from the invalid client_ID or Client_Secret " +
+                    "which are used to get the OAuth response from the URL : " + accessRequest.getLocationUri() +
+                    " Check the client_Secret value that associate with client_ID. ", e);
+        } catch (OAuthSystemException e) {
+            throw new AuthenticationFailedException("Communication errors between the OAuth Recourse Server and " +
+                    "the OAuth Authorization Server while getting the OAuth response form the URL: " +
+                    accessRequest.getLocationUri(), e);
+        }
+        return oAuthResponse;
+    }
+
+    /**
+     * Get the access token from endpoint with generated code.
+     *
+     * @param tokenEndPoint the Access_token endpoint.
+     * @param clientId      the Client ID.
+     * @param code          the Code.
+     * @param clientSecret  the Client Secret.
+     * @param callbackurl   the CallBack URL.
+     * @return the access token
+     */
+    private OAuthClientRequest getAccessRequest(String tokenEndPoint, String clientId, String code, String clientSecret,
+                                                String callbackurl) throws AuthenticationFailedException {
+
+        OAuthClientRequest accessRequest;
+        try {
+            accessRequest = OAuthClientRequest.tokenLocation(tokenEndPoint).setGrantType(GrantType.AUTHORIZATION_CODE)
+                    .setClientId(clientId).setClientSecret(clientSecret).setRedirectURI(callbackurl).setCode(code)
+                    .buildBodyMessage();
+        } catch (OAuthSystemException e) {
+            throw new AuthenticationFailedException("Communication errors occurred between the OAuth Recourse Server " +
```

Comment:
This shouldn't be due to a 'Communication error'. The error occurs while building the token request. Just mention in the error something like, 'Error occurred while building the token request with the parameter ......'